packages/block-editor/src/components/contrast-checker/index.js

```diff
@@ -34,6 +35,11 @@ function ContrastChecker( {
 	const msg = tinyBackgroundColor.getBrightness() < tinyTextColor.getBrightness() ?
 		__( 'This color combination may be hard for people to read. Try using a darker background color and/or a brighter text color.' ) :
 		__( 'This color combination may be hard for people to read. Try using a brighter background color and/or a darker text color.' );
+	if ( tinyBackgroundColor.getBrightness() < tinyTextColor.getBrightness() ) {
+		speak( __( 'This color combination may be hard for people to read. Try using a darker background color and/or a brighter text color.' ) );
+	} else {
+		speak( __( 'This color combination may be hard for people to read. Try using a brighter background color and/or a darker text color.' ) );
+	}
```

Comment:
Hi there! thanks for the PR, I think we shouldn't call `speak` (or any sideeffect) directly in render functions because these can be called over and over again when the component rerenders.

A solution here might be to wrap these in `useEffect` React hook and only call this once this `tinyBackgroundColor.getBrightness() < tinyTextColor.getBrightness()` changes.

Reply:
More details about `useEffect` here:  https://reactjs.org/docs/hooks-effect.html

Reply:
@youknowriad Thank you for pointing that out! I have updated my code accordingly. I am passing in the color values as the second argument so that speak is called again whenever someone changes the color options (I am not passing in msg, as the content of the error message may be the same even if the user picks different colors so speak wouldn't be called again even though a warning would be displayed)